File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        List<ListNode> list = new LinkedList<>();
        ListNode slow = head; 
        while (slow != null) {
            list.add(slow);
            slow = slow.next;
        }
        return list.get(list.size()/2);
    }
}

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        keepNode=[]
        dummy=head
        while dummy:
            keepNode.append(dummy)
            dummy=dummy.next
        N=len(keepNode)
        fm=N//2
        d=head
        while fm:
            fm-=1
            d=d.next 
        return d
        